Build Error: Minecraft 1.21.11

## Build Error Report

**Minecraft Version:** 1.21.11
**Commit:** 2dc7e6d
**Error:** ShellError: Failed with exit code 1

### Full Build Logs

```
ShellError: Failed with exit code 1

To honour the JVM settings for this build a single-use Daemon process will be forked. For more on this, please refer to https://docs.gradle.org/9.2.1/userguide/gradle_daemon.html#sec:disabling_the_daemon in the Gradle documentation.
Daemon will be stopped at the end of the build 

> Configure project :
Fabric Loom: 1.14.10
:remapping sources

> Task :compileJava
/tmp/mod-vc0or2ylp4/src/main/java/me/imgalvin/bansleeping/BanSleeping.java:14: error: cannot find symbol
			return PlayerEntity.SleepFailureReason.OTHER_PROBLEM;
			                                      ^
  symbol:   variable OTHER_PROBLEM
  location: class SleepFailureReason
1 error

> Task :compileJava FAILED

[Incubating] Problems report is available at: file:///tmp/mod-vc0or2ylp4/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> Compilation failed; see the compiler output below.
  /tmp/mod-vc0or2ylp4/src/main/java/me/imgalvin/bansleeping/BanSleeping.java:14: error: cannot find symbol
  			return PlayerEntity.SleepFailureReason.OTHER_PROBLEM;
  			                                      ^
    symbol:   variable OTHER_PROBLEM
    location: class SleepFailureReason
  1 error

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (powered by Develocity).

BUILD FAILED in 1m 7s
1 actionable task: 1 executed

```

---
*This issue was automatically created by the Minecraft CI system.*

--- a/README.md
+++ b/README.md
@@ -2,10 +2,18 @@
 Ban Sleeping is a Minecraft Mod for Fabric 1.21 that prevents the player from sleeping in a bed
 
 # Compatible Versions
+[![Supported versions](https://modrinth-shields.imgalvin.me/badge/bansleeping)](https://modrinth.com/mod/bansleeping)
+
+## New system
+Due to the new versioning system, each supported game version has its own mod version file. Please look at the versions badge above for more
+
+## Old system
+*Only applicable for 1.21.10 and below*
+
 | Mod Version | Loader | Game Version(s) | Mod Version File        |
 |-------------|--------|-----------------|-------------------------|
-| 1.0.1       | Fabric | 1.21            | 1.0.1-fabric+1.21.jar   |
-| 1.0.0       | Fabric | 1.21            | 1.0.0-fabric+1.21.jar   |
+| 1.0.1       | Fabric | 1.21-1.21.10    | 1.0.1-fabric+1.21.jar   |
+| 1.0.0       | Fabric | 1.21-1.21.5     | 1.0.0-fabric+1.21.jar   |
 
 # Important Note
 This mod is a **server-side** mod!  

--- a/README.md
+++ b/README.md
@@ -1,11 +1,15 @@
 # Ban Sleeping
-Ban Sleeping is a Minecraft Mod for Fabric 1.21 that prevents the player from sleeping in a bed
+Ban Sleeping is a Minecraft Mod for Fabric 1.20+ that prevents the player from sleeping in a bed
 
 # Compatible Versions
 [![Supported versions](https://modrinth-shields.imgalvin.me/badge/bansleeping)](https://modrinth.com/mod/bansleeping)
 
 ## New system
-Due to the new versioning system, each supported game version has its own mod version file. Please look at the versions badge above for more
+Due to the new Minecraft versioning system and their "drops", each supported game version has its own mod version file to reduce load and potential problems. Please look at the versions badge above for more.
+
+Also, the minimum loader and API versions have been standardised to be the minimum available for that Minecraft version. This is to prevent crashes from new updates; but very few people are to run into this issue as most people update their loaders and APIs frequently.
+
+I recommend heavily using a mod manager/launcher to help automatically update these dependencies. Modrinth and Prism Launcher are great options for this.
 
 ## Old system
 *Only applicable for 1.21.10 and below*
@@ -22,7 +26,11 @@ Playing on a **server**? You are not required to have this downloaded.
 
 Also: This mod requires the **Fabric API**
 
-## About 1.0.1
+# Changelog
+## 1.0.2 (1.21.11)
+- Updated required loader version to 0.17.3
+
+## 1.0.1
 1.0.1 was just a refactoring change, no new features were added
 
 # Demonstration of Mod

--- a/gradle.properties
+++ b/gradle.properties
@@ -9,7 +9,7 @@ org.gradle.configuration-cache=false
 # check these on https://fabricmc.net/develop
 minecraft_version=1.21.11
 yarn_mappings=1.21.11+build.3
-loader_version=0.18.4
+loader_version=0.17.3
 loom_version=1.14-SNAPSHOT
 
 # Fabric API

--- a/gradle.properties
+++ b/gradle.properties
@@ -7,15 +7,15 @@ org.gradle.configuration-cache=false
 
 # Fabric Properties
 # check these on https://fabricmc.net/develop
-minecraft_version=1.21
-yarn_mappings=1.21+build.9
-loader_version=0.15.11
+minecraft_version=1.21.11
+yarn_mappings=1.21.11+build.3
+loader_version=0.18.4
 loom_version=1.14-SNAPSHOT
 
 # Fabric API
-fabric_version=0.102.0+1.21
+fabric_version=0.140.2+1.21.11
 
 # Mod Properties
-mod_version=1.0.1+1.21
+mod_version=1.0.1+1.21.11
 maven_group=me.imgalvin.bansleeping
 archives_base_name=bansleeping

--- a/gradle.properties
+++ b/gradle.properties
@@ -16,6 +16,6 @@ loom_version=1.14-SNAPSHOT
 fabric_version=0.140.2+1.21.11
 
 # Mod Properties
-mod_version=1.0.1+1.21.11
+mod_version=1.0.2+1.21.11
 maven_group=me.imgalvin.bansleeping
 archives_base_name=bansleeping

--- a/src/main/java/me/imgalvin/bansleeping/BanSleeping.java
+++ b/src/main/java/me/imgalvin/bansleeping/BanSleeping.java
@@ -9,9 +9,6 @@ import net.minecraft.util.math.BlockPos;
 public class BanSleeping implements ModInitializer {
 	@Override
 	public void onInitialize() {
-		EntitySleepEvents.ALLOW_SLEEPING.register((PlayerEntity player, BlockPos blockPos) -> {
-			player.sendMessage(Text.of("Sleeping is banned!"), true);
-			return PlayerEntity.SleepFailureReason.OTHER_PROBLEM;
-		});
+		EntitySleepEvents.ALLOW_SLEEPING.register((PlayerEntity player, BlockPos blockPos) -> new PlayerEntity.SleepFailureReason(Text.literal("Sleeping is banned!")));
 	}
 }

--- a/src/main/resources/fabric.mod.json
+++ b/src/main/resources/fabric.mod.json
@@ -21,9 +21,9 @@
 		]
 	},
 	"depends": {
-		"fabricloader": ">=0.15.11",
-		"minecraft": "~1.21",
+		"fabricloader": ">=0.17.3",
+		"minecraft": "1.21.11",
 		"java": ">=21",
-		"fabric-api": "*"
+		"fabric-api": ">=0.140.0"
 	}
 }

